src/enrichment_tables/geoip.rs

```diff
@@ -0,0 +1,413 @@
+                        subdivision
+                            .names
+                            .as_ref()
+                            .and_then(|names| names.get(&*self.config.locale))
+                    })
+                    .map(|&name| name.into()),
+            );
+            add_field(
+                "region_code",
+                subdivision
+                    .and_then(|subdivision| subdivision.iso_code)
+                    .map(Into::into),
+            );
+
+            add_field(
+                "postal_code",
+                data.postal.and_then(|p| p.code).map(Into::into),
+            );
+        }
+
+        Some(map)
+    }
+}
+
+impl Table for Geoip {
+    /// Search the enrichment table data with the given condition.
+    /// All conditions must match (AND).
+    ///
+    /// # Errors
+    /// Errors if no rows, or more than 1 row is found.
+    fn find_table_row<'a>(
+        &self,
+        case: Case,
+        condition: &'a [Condition<'a>],
+        select: Option<&[String]>,
+        index: Option<IndexHandle>,
+    ) -> Result<BTreeMap<String, Value>, String> {
+        self.find_table_rows(case, condition, select, index)?
+            .pop()
+            .ok_or_else(|| "IP not found".to_string())
+    }
+
+    /// Search the enrichment table data with the given condition.
+    /// All conditions must match (AND).
+    /// Can return multiple matched records
+    fn find_table_rows<'a>(
+        &self,
+        _: Case,
+        condition: &'a [Condition<'a>],
+        select: Option<&[String]>,
+        _: Option<IndexHandle>,
+    ) -> Result<Vec<BTreeMap<String, Value>>, String> {
+        match condition.get(0) {
+            Some(_) if condition.len() > 1 => Err("Only one condition is allowed".to_string()),
+            Some(Condition::Equals { value, .. }) => {
+                let ip = value
+                    .to_string_lossy()
+                    .parse::<IpAddr>()
+                    .map_err(|_| "Invalid IP address".to_string())?;
+                Ok(self
+                    .lookup(ip, select)
+                    .map(|values| vec![values])
+                    .unwrap_or_default())
+            }
+            Some(_) => Err("Only equality condition is allowed".to_string()),
+            None => Err("IP condition must be specified".to_string()),
+        }
+    }
+
+    /// Hints to the enrichment table what data is going to be searched to allow it to index the
+    /// data in advance.
+    ///
+    /// # Errors
+    /// Errors if the fields are not in the table.
+    fn add_index(&mut self, case: Case, fields: &[&str]) -> Result<IndexHandle, String> {
+        match fields.len() {
+            0 => Err("IP field is required".to_string()),
+            1 => {
+                let index = IndexHandle(self.indexes.len());
+                self.indexes
+                    .push((case, fields.iter().map(|field| field.to_string()).collect()));
+                Ok(index)
```

Comment:
```suggestion
                Ok(0)
```

It's still useful to keep the error checks here, but we can remove the `push`.